TwitchRSS/twitchrss.py

```diff
@@ -128,14 +130,25 @@ def fetch_user(channel_name):
 
 @cached(cache=TTLCache(maxsize=1000, ttl=VODCACHE_LIFETIME))
 def fetch_vods(channel_id):
-    return fetch_json(channel_id, VOD_URL_TEMPLATE)
+    clipFilter = request.args.get('filter')
+    
+    if clipFilter not in VALID_URL_ARGS:
+        clipFilter = 'all'
+    if clipFilter is None:
+        clipFilter = 'all'
+
+    return fetch_json(channel_id, VOD_URL_TEMPLATE, clipFilter)
```

Comment:
This function is `@cached` so subsequent calls will just return the previous result if the parameters are the same.
So if I fetch for `channel_id=foo` and `clipFilter=all` and then same `channel_id` but `clipFilter=archive` it will return the `all` result since the key for the cache is just `foo`. If you pass the `clipFilter` as a parameter then the cache will work since it will take both id and filter as the combined key.